src/schema/v2/quiz.ts

```diff
@@ -34,6 +35,55 @@ export const QuizType = new GraphQLObjectType<any, ResolverContext>({
         }
       },
     },
+    recommendedArtworks: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(ArtworkType))
+      ),
+      resolve: async (
+        { quiz_artworks },
+        _,
+        { savedArtworkLoader, relatedLayerArtworksLoader }
+      ) => {
+        if (!savedArtworkLoader) return []
+
+        try {
+          const artworks = await Promise.all(
+            quiz_artworks.map(({ artwork_id }) => {
+              return savedArtworkLoader(artwork_id)
+            })
+          )
+
+          const savedArtworks = artworks.filter(({ is_saved }) => is_saved)
+
+          if (savedArtworks.length === 0) return []
+
+          const recommendedArtworks = await Promise.all(
+            savedArtworks.map(({ id }) => {
+              return relatedLayerArtworksLoader(
+                { id: "main", type: "synthetic" },
+                { artwork: [id] },
+                { requestThrottleMs: 86400000 }
+              )
+            })
+          )
+
+          // Rather than pass a `size` to the `relatedLayerArtworksLoader`
+          // we filter results after the fact rather than split the cache across
+          // different sizes.
```

Comment:
So note here rather than splitting the cache into individual keys for each `size` param, I'm just requesting _all_ of the results and filtering it down from there.